detect_secrets/plugins/common/filters.py

```diff
@@ -97,3 +78,79 @@ def _is_sequential_string(secret, *args):
             return True
 
     return False
+
+
+ALL_FALSE_POSITIVE_HEURISTICS = (
+    is_found_with_aho_corasick,
+    is_sequential_string,
+)
+
+
+# NOTE: this doesn't handle key-values on a line properly.
+# NOTE: words that end in "id" will be treated as ids
+_ID_DETECTOR_REGEX = re.compile(r'[iI][dD][^A-Za-z0-9]')
+
+
+def is_likely_id_string(secret, line):
+    """
+    :type secret: str
+
+    :type line: str
+    :param line: Line context for the plaintext secret
+
+    :rtype: bool
+    Returns true if the secret could be an id, false otherwise.
+    """
+    if secret not in line:
+        return False
+
+    secret_index = line.index(secret)
+    return _ID_DETECTOR_REGEX.findall(line, pos=0, endpos=secret_index)
+
+
+ALL_FALSE_POSITIVE_WITH_LINE_CONTEXT_HEURISTICS = [
+    is_likely_id_string,
```

Comment:
From previous iteration tests, (running it on open-source python repos, adding heuristics to give better [signal stats from `--display-results` output](https://github.com/Yelp/detect-secrets/pull/241/commits/433b75e82347b1c806af3723ae2d8f3c70fc3827)), we added a few heuristics [1] we can add here.

We can add a UUID check when calling `is_false_positive` from `high_entropy_strings.py` and a "value by itself on a line" check to `is_false_positive_with_line_context`. The latter being when files had
```python
some_list_of_ids = [
     'frequently high entropy but not sensitive',
]
```






[1] Hacky code only used in a one-off branch to improve signal, not meant to be merged as written
```python
            rest_of_the_line = string.replace(result, '').lower().strip()
            if any(
                false_positive in rest_of_the_line
                for false_positive in (
                    'id',          # Often not of value
                    're.match',    # Regular expressions have high-entropy
                    're.compile',  # Regular expressions have high-entropy
                    '<script>',    # Often not of value
                )
            ):
                continue

            if len(result) == 36:
                UUID_RE = re.compile(
                    r'^[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89AB][0-9a-f]{3}-[0-9a-f]{12}$',
                    re.IGNORECASE,
                )
                if UUID_RE.match(result):
                    continue

            if is_false_positive(result, self.automaton):
                continue

            # Often times IDs are listed in a list.
            # If the secret is on a line alone by itself,
            # it is likely not to be of value
            # I went from 15% signal to 41% with this on one test corpus
            BY_ITSELF_OPTIONAL_COMMENT = re.compile(
                r'^(b)?(\'\'|""),\s*?#',
                re.IGNORECASE,
            )
            if BY_ITSELF_OPTIONAL_COMMENT.match(rest_of_the_line):
                continue
```

Reply:
That UUID regex may only be for certain kinds of UUIDs, we can always make it better, I just wanted to say that explicitly in case you wanted to improve it, or if it didn't work for a test UUID.

Reply:
Aside from the heuristics above, I'd like some data backing up that this does improve signal, via [1], otherwise we don't really know if it does what we intend it to do.


[1]
1. Run it on a corpus
1. Run `--display-results`
1. Change the plugin to trim what are clearly false-positives
1. Output Baseline A (only running this plugin)
1. Audit Baseline A
1. Try to trim down false-positives even more
1. Output a new baseline B (only running this plugin)
1. (Err, we don't have code in detect-secrets right now to propagate `is_secret` decisions from Baseline A to Baseline B, I had to write a hacky script to do it, I can share it.)
1. Compare how much of a signal improvement there is
1. See if signal is above 40%
1. Repeat process with new corpus

Reply:
I'm a bit ambivalent on ignoring UUIDs. Based on some googling there's at least some usage of UUIDs as secrets in the wild.

Reply:
Actually I take that back, looks like I misunderstood the docs I was reading.